File: highlight/src/main/java/com/neo/highlight/core/Scheme.java
package com.neo.highlight.core;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.regex.Pattern;

/**
 * @author Irineu A. Silva
 */
public interface Scheme {

    @Nullable
    Pattern getRegex();

    @Nullable
    Object getSpan(@NonNull ScopeResult scopeResult);

    boolean getClearOldSpan();

    @NonNull
    Scheme setClearOldSpan(boolean clearOldSpan);

}

File: highlight/core/src/commonMain/kotlin/com/neoutils/highlight/core/Scheme.kt
package com.neoutils.highlight.core

import com.neoutils.highlight.core.util.Matcher
import com.neoutils.xregex.XRegex

interface Scheme<T : Any> {
    val regex: XRegex
    val matcher: Matcher<T>
    val range: IntRange?
}
